src/rdf.rs

```diff
@@ -0,0 +1,277 @@
+            map_format_to_head(fmt_res.map_or("".to_string(), |fmt| fmt.object.to_string()))
+        });
+
+    for acc_url_result in list_access_urls(dist_node, store) {
+        match acc_url_result {
+            Ok(acc_url) => match acc_url.object {
+                Term::NamedNode(acc_url_node) => {
+                    urls.push(UrlCheck {
+                        method: head.to_string(),
+                        url: acc_url_node.into_string(),
+                        url_type: UrlType::AccessUrl,
+                    });
+                }
+                node => warn!("Access URL node is not a NamedNode but {}", node),
+            },
+            Err(e) => return Err(e.into()),
+        }
+    }
+
+    for dl_url_result in list_download_urls(dist_node, store) {
+        match dl_url_result {
+            Ok(dl_url) => match dl_url.object {
+                Term::NamedNode(dl_url_node) => {
+                    urls.push(UrlCheck {
+                        method: head.to_string(),
+                        url: dl_url_node.into_string(),
+                        url_type: UrlType::DownloadUrl,
+                    });
+                }
+                node => warn!("Download URL node is not a NamedNode but {}", node),
+            },
+            Err(e) => return Err(e.into()),
+        }
+    }
+
+    Ok(urls)
+}
+
+// Create new memory metrics store for supplied dataset
+pub fn create_metrics_store(dataset: NamedNodeRef) -> Result<Store, Error> {
+    let store = Store::new()?;
+
+    // Insert dataset
+    store.insert(
+        Quad::new(
+            dataset.clone(),
+            rdf::TYPE,
+            dcat::DATASET_CLASS,
+            GraphName::DefaultGraph,
+        )
+        .as_ref(),
+    )?;
+    Ok(store)
+}
+
+// Add MQA url status code metric
+pub fn add_url_status_metric(
+    distribution: NamedOrBlankNodeRef,
+    url: NamedNodeRef,
+    url_type: UrlType,
+    status_code: u16,
+    store: &Store,
+) -> Result<BlankNode, Error> {
+    match url_type {
+        UrlType::AccessUrl => {
+            store.insert(
+                Quad::new(distribution, dcat::ACCESS_URL, url, GraphName::DefaultGraph).as_ref(),
+            )?;
+
+            add_quality_measurement(
+                dcat_mqa::ACCESS_URL_STATUS_CODE,
+                distribution,
+                url.into(),
+                status_code,
+                store,
+            )
+        }
+        UrlType::DownloadUrl => {
+            store.insert(
+                Quad::new(
+                    distribution,
+                    dcat::DOWNLOAD_URL,
+                    url,
+                    GraphName::DefaultGraph,
+                )
+                .as_ref(),
+            )?;
+
+            add_quality_measurement(
+                dcat_mqa::DOWNLOAD_URL_STATUS_CODE,
+                distribution,
+                url.into(),
+                status_code,
+                store,
+            )
+        }
+    }
```

Comment:
Foreslo å merge de to funksjonene for å unngå å skrive alt dobbelt opp.

```suggestion
    let (predicate, metric) = match url_type {
        UrlType::AccessUrl => (dcat::ACCESS_URL, dcat_mqa::ACCESS_URL_STATUS_CODE),
        UrlType::DownloadUrl => (dcat::DOWNLOAD_URL, dcat_mqa::DOWNLOAD_URL_STATUS_CODE),
    };

    store.insert(Quad::new(distribution, predicate, url, GraphName::DefaultGraph).as_ref())?;
    add_quality_measurement(metric, distribution, url.into(), status_code, store)
```